src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -334,88 +340,107 @@ public async Task<AuthenticationResult> GetAuthenticationResultForUserAsync(
         {
+            string? username = null;
+            string? password = null;
+            string? agentIdentity = string.Empty;
+
+            // Case where the user is passed through the Claims identity
             if (user != null && user.HasClaim(c => c.Type == ClaimConstants.Username) && user.HasClaim(c => c.Type == ClaimConstants.Password))
             {
-                string username = user.FindFirst(ClaimConstants.Username)?.Value ?? string.Empty;
-                string password = user.FindFirst(ClaimConstants.Password)?.Value ?? string.Empty;
-                bool forceRefresh = tokenAcquisitionOptions?.ForceRefresh ?? false;
+                username = user.FindFirst(ClaimConstants.Username)?.Value ?? string.Empty;
+                password = user.FindFirst(ClaimConstants.Password)?.Value ?? string.Empty;
+            }
+
+            // Case of the Agent User identities
+            var extraParameters = tokenAcquisitionOptions?.ExtraParameters;
+            if (extraParameters != null && extraParameters.ContainsKey(Constants.AgentIdentityKey) && extraParameters.ContainsKey(Constants.UsernameKey))
+            {
+                // If the agentId is present, we can use it
+                username = extraParameters[Constants.UsernameKey] as string;
+                agentIdentity = extraParameters[Constants.AgentIdentityKey] as string;
+                password = "password";
+            }
+
+            if (username == null)
+            {
+                return null;
+            }
+
+            bool forceRefresh = tokenAcquisitionOptions?.ForceRefresh ?? false;
 
-                if (!forceRefresh && user.GetMsalAccountId() != null)
+            if (!forceRefresh && user != null && user.GetMsalAccountId() != null)
+            {
+                try
                 {
-                    try
-                    {
-                        var account = await application.GetAccountAsync(user.GetMsalAccountId()).ConfigureAwait(false);
-
-                        // Silent flow
-                        return await application.AcquireTokenSilent(
-                            scopes.Except(_scopesRequestedByMsal),
-                            account)
-                            .ExecuteAsync()
-                            .ConfigureAwait(false);
-                    }
-                    catch (MsalException ex)
-                    {
-                        // Log a message when the silent flow fails and try acquisition through ROPC.
-                        Logger.TokenAcquisitionError(_logger, ex.Message, ex);
-                    }
+                    var account = await application.GetAccountAsync(user.GetMsalAccountId()).ConfigureAwait(false);
 
+                    // Silent flow
+                    return await application.AcquireTokenSilent(
```

Comment:
When I ran the test - AgentUserIdentityGetsTokenForGraphAsync, it did not hit the AcquireTokenSilent path (I changed it to request 2 tokens). The user is always null here.